### PR TITLE
Fix netty-all artifact

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

612ab584489eae74b28ce0bc1b75fb23c5d24b57 did change the way how netty-all was produced by unfortunally it messed up the dependency graph for our native artifacts. This commit reverts changes done by 612ab584489eae74b28ce0bc1b75fb23c5d24b57 and also clean up the profiles

Modifications:

netty-all is useable again

Result:

Fixes https://github.com/netty/netty/issues/11272

### CODE REVIEW

File: all/pom.xml [hunk 1]
```diff
@@ -232,250 +249,395 @@
...
           <groupId>${project.groupId}</groupId>
           <artifactId>netty-transport-native-epoll</artifactId>
           <scope>compile</scope>
+          <optional>true</optional>
         </dependency>
       </dependencies>
     </profile>
-
-    <profile>
-      <id>full</id>
```
Comment (normanmaurer) [inline all/pom.xml:251]: This is not needed anymore as we moved the Javadoc etc generated to the parent pom.xml

**Comment by fredericBregier** [comment]
@normanmaurer OK, I will wait for a new release to be able to check, I think it shall be ok. Thanks !

**Review by chrisvest — APPROVED** [review]
I guess this reopens #4671

**Comment by normanmaurer** [comment]
@chrisvest sadly yes :/

**Comment by candrews** [comment]
Instead of reverting, I feel like fixing the problem would be preferable. I'll take a look - I see others are doing so as well

**Comment by normanmaurer** [comment]
@candrews generally speaking yes... that said I think we need to release a regression fix release asap. And as it has shown we  need to take more time to really ensure that everything works as expected.

**Comment by chrisvest** [comment]
I agree with @normanmaurer. Since this turned out to be unexpectedly tricky, and there are other fixes waiting for release, I think it's a better trade-off to do the revert in this PR, regressing the netty-all class path behaviour to the state it's been for years prior, and then there's no time pressure on #11278.

**Comment by normanmaurer** [comment]
Let's pull this in and then #11278 can revert this change and add its own changes on top of it.

**Pull request merged**